From: Kentaro Hayashi <kenhys@gmail.com>
Date: Mon, 10 Mar 2025 20:43:37 +0900
Subject: Fix FTBFS with absl 20240722.0
MIME-Version: 1.0
Content-Type: text/plain; charset="utf-8"
Content-Transfer-Encoding: 8bit

Description: Fix the following error:
  FAILED: obj.host/dictionary/file/codec.codec.o /usr/bin/g++ -MMD -MF
  obj.host/dictionary/file/codec.codec.o.d -DMOZC_GYP_BUILD -DOS_LINUX
  -DMOZC_BUILD -DCHANNEL_DEV -DENABLE_GTK_RENDERER -DNDEBUG
  -DQT_NO_DEBUG -DMOZC_NO_LOGGING -DIGNORE_HELP_FLAG
  -DIGNORE_INVALID_FLAG -I../../third_party/abseil-cpp
  '-I/build/reproducible-path/mozc-2.29.5160.102+dfsg/src' -Igen
  -Igen/proto_out -fmessage-length=0 -fno-strict-aliasing
  -funsigned-char -pipe -pthread -fno-omit-frame-pointer
  -fstack-protector --param=ssp-buffer-size=4 -Wall -Wno-char-subscripts
  -Wno-sign-compare -Wno-deprecated-declarations -Wwrite-strings
  -Wno-unknown-warning-option -Wno-inconsistent-missing-override -fPIC
  -fno-exceptions -O2 -Wno-deprecated -c ../../dictionary/file/codec.cc
  -o obj.host/dictionary/file/codec.codec.o
  ../../dictionary/file/codec.cc: In member function ‘virtual
  absl::debian7::Status
  mozc::dictionary::DictionaryFileCodec::ReadSections(const char*, int,
  std::vector<mozc::dictionary::DictionaryFileSection>*) const’:
  ../../dictionary/file/codec.cc:121:15: error: ‘StrCat’ is not a
  member of ‘absl’ 121 | absl::StrCat("codec.cc: Insufficient data
  size: ", length, " bytes")); | ^~~~~~
  ../../dictionary/file/codec.cc:126:15: error: ‘StrCat’ is not a
  member of ‘absl’ 126 | absl::StrCat("codec.cc: memory block of size
  ", length, | ^~~~~~
Author: Kentaro Hayashi <kenhys@xdump.org>
Forwarded: not-needed
Last-Update: 2025-03-10
---
 src/dictionary/file/codec.cc | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/dictionary/file/codec.cc b/src/dictionary/file/codec.cc
index 6fc3a36..a1e9bc6 100644
--- a/src/dictionary/file/codec.cc
+++ b/src/dictionary/file/codec.cc
@@ -42,6 +42,7 @@
 #include "dictionary/file/codec_util.h"
 #include "dictionary/file/section.h"
 #include "absl/status/status.h"
+#include "absl/strings/str_cat.h"
 #include "absl/strings/string_view.h"
 
 namespace mozc {
